### PR TITLE
Two DQM analyzer for Pixel Digi and Clusters produced by either GPU or CPU

### PR DESCRIPTION
**Pull request by ttedeschi**
#### PR description:
This PR introduces two DQM analyzer for Pixel Digi and Clusters produced by either GPU or CPU. It is a follow up of #35965.

#### PR validation:
Tested with 11634.501 wf for CPU 11634.506 wf for GPU

#### if this PR is a backport please specify the original PR and why you need to backport that PR:
no backport needed

### CODE REVIEW

**Review by mmusich — COMMENTED** [review]
I am not sure to understand the purpose of the new modules, `SiPixelPhase1MonitorClusters` and `SiPixelPhase1MonitorDigis` within the `DQM/SiPixelPhase1Heterogeneous` package.
These modules seem to look at regular old plain legacy data-structures `edm::DetSetVector<T>>` instead of `SoA` quantities, so they would be *de facto* redundant duplicates of the regular pixel phase-1 DQM, could you elaborate why this layout was chosen?

**Comment by jfernan2** [comment]
enable gpu

File: DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorClusters.cc [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    uint32_t nClustersEndcapDiskp1 = 0;
+    uint32_t nClustersEndcapDiskp2 = 0;
+    uint32_t nClustersEndcapDiskp3 = 0;
+    for (it = input->begin(); it != input->end(); ++it) {
+      const uint32_t nClustersEv = it->size();
+      nClusters += nClustersEv;
+      DetId id = it->detId();
+      uint32_t subdetid = (id.subdetId());
+        if (subdetid == PixelSubdetector::PixelBarrel) {
+            edm::LogWarning("SiPixelPhase1MonitorClusters") << " PX Barrel:  DetId " <<id.rawId()<<" Layer "<<trackerTopology_->pxbLayer(id)<<std::endl;
```
[the remark this answers is not in the document]
Comment (ttedeschi) [reply DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorClusters.cc:147]: only used for debugging porpouse, I deleted it

File: DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorClusters.cc [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+  MonitorElement* hClustersEndcapDiskp2Size;
+  MonitorElement* hClustersEndcapDiskp3Size;
+};
+
+//
+// constructors
+//
+
+SiPixelPhase1MonitorClusters::SiPixelPhase1MonitorClusters(const edm::ParameterSet& iConfig) {
+  tokenCluster_ = consumes<SiPixelClusterCollectionNew>(iConfig.getParameter<edm::InputTag>("pixelClusters"));
```
Comment (mmusich) [inline DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorClusters.cc:104]: now that they are declared `const` these should be in the class initializer list.

File: DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorClusters.cc [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+//
+// -- Analyze
+//
+void SiPixelPhase1MonitorClusters::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  // get geometry
+  edm::ESHandle<TrackerGeometry> tracker = iSetup.getHandle(trackerGeomToken_);
+  assert(tracker.isValid());
+  // TrackerTopology for module informations
+  edm::ESHandle<TrackerTopology> trackerTopologyHandle = iSetup.getHandle(trackerTopoToken_);
+  trackerTopology_ = trackerTopologyHandle.product();
```
Comment (mmusich) [inline DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorClusters.cc:119]:
```suggestion
  trackerTopology_ = &iSetup.getData(trackerTopoToken_);
```

File: DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorDigis.cc [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+//
+// -- Analyze
+//
+void SiPixelPhase1MonitorDigis::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  // get geometry
+  edm::ESHandle<TrackerGeometry> tracker = iSetup.getHandle(trackerGeomToken_);
+  assert(tracker.isValid());
+  // TrackerTopology for module informations
+  edm::ESHandle<TrackerTopology> trackerTopologyHandle = iSetup.getHandle(trackerTopoToken_);
+  trackerTopology_ = trackerTopologyHandle.product();
```
Comment (mmusich) [inline DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorDigis.cc:102]:
```suggestion
  trackerTopology_ = &iSetup.getData(trackerTopoToken_);
```

**Comment by silviodonato** [comment]
kind reminder @ttedeschi 
the last open pre-release of 12_3_X will be cut in 2-3 weeks from now!

**Pull request closed without merging**